src/main/java/org/jvnet/hudson/test/LoggerRule.java

```diff
@@ -78,7 +79,7 @@ public LoggerRule() {
      * @return this rule, for convenience
      */
     public LoggerRule capture(int maximum) {
-        messages = new ArrayList<String>();
+        messages = Collections.synchronizedList(new ArrayList<>());
```

Comment:
Or just synchronize manually in the one piece of code which adds to it; since you are manually synchronizing when copying it, `synchronizedList` does not really simplify anything.